doc/movingparts.rst

```diff
@@ -55,17 +52,16 @@ The implementation of builders can be found in `html5lib/treebuilders/
 Tree walkers
 ------------
 
-Once a tree is ready, you can work on it either manually, or using
-a tree walker, which provides a streaming view of the tree. html5lib
-provides walkers for all three supported types of trees (``etree``,
-``dom`` and ``lxml``).
+In addition to manipulating a tree directly, you can use a tree walker to generate a streaming view of it.
+html5lib provides walkers for ``etree``, ``dom``, and ``lxml`` trees, as well as ``genshi`` `markup streams <https://genshi.edgewall.org/wiki/Documentation/streams.html>`_.
 
 The implementation of walkers can be found in `html5lib/treewalkers/
 <https://github.com/html5lib/html5lib-python/tree/master/html5lib/treewalkers>`_.
 
-Walkers make consuming HTML easier. html5lib uses them to provide you
-with has a couple of handy tools.
+html5lib provides a few tools for consuming token streams:
 
+* :class:`~html5lib.serializer.HTMLSerializer`, to generate a stream of bytes; and
+* filters, to manipulate the token stream.
```

Comment:
The leader has "a few tools", but this list has two items. Further, the items seem like a sentence. I'd either change the bullet list into a sentence or unsentencify the bullet items. Maybe something like this?:

```
html5lib provides two tools for consuming token streams:

* :class:`~html5lib.serializer.HTMLSerializer` for generating a stream of bytes
* filters for manipulating the token stream
```

Reply:
I ended up going with a sentence, rather than the bulleted list, as there aren't exactly two (really there's the serializer and a bunch of filters), so it's really two _categories_ of token consumer, but that distinction isn't really useful to point out.